### PR TITLE
[Parse] 'where' to ',' fix-it should remove a preceding space.

### PR DESCRIPTION
**Pull request by jrose-apple**
rdar://problem/27709302

### CODE REVIEW

**Comment by jrose-apple** [comment]
@jtbandes, you've been doing some Parse work. Mind double-checking this for me to cherry-pick to the swift-3.0-branch?

**Comment by jrose-apple** [comment]
@swift-ci Please smoke test

File: lib/Parse/ParseStmt.cpp [hunk 1]
```diff
@@ -1179,17 +1179,18 @@ ParserStatus Parser::parseStmtCondition(StmtCondition &Condition,
...
       diagnose(Tok, diag::expected_comma_stmtcondition)
-      .fixItReplace(Tok.getLoc(), ",");
+        .fixItReplaceChars(getEndOfPreviousLoc(), Tok.getRange().getEnd(), ",");
       consumeToken();
       return true;
     }
 
     // Boolean conditions are separated by commas, not the 'where' keyword, as
     // they were in Swift 2 and earlier.
     SourceLoc whereLoc;
```
Comment (jtbandes) [inline lib/Parse/ParseStmt.cpp]: Is whereLoc unused now?

**Review by jtbandes — COMMENTED** [review]
This looks fine to me, but I wonder what @rintaro thinks too — perhaps the work from #4499 could be extended to handle cases like this. I don't think there are any commas we'd want to insert while leaving whitespce before them.

File: lib/Parse/ParseStmt.cpp (hunk 1 above)
Comment (jrose-apple) [inline lib/Parse/ParseStmt.cpp]: Ah, so it is. Thanks!

**Comment by jrose-apple** [comment]
@swift-ci Please smoke test

**Comment by jrose-apple** [comment]
I think I'm going to merge this anyway and move forwards. Rintaro, if you have any suggestions, I'm happy to do it as follow-up.

**Comment by jrose-apple** [comment]
(In particular it seems like you'd have to look arbitrarily far back to delete whitespace, and that might be too much magic to do in general.)

**Pull request merged**